[Core] TypeScript `type` import error `1.5.1`

Hello Team,
We are using react v 18.2.0 for our Frond end application and we have deleted node-modules and again run npm install, after that i have run npm run build. but we are getting below complied error:
![image (2)](https://github.com/floating-ui/floating-ui/assets/64414384/2a9bfe99-7be8-424d-9a94-7319ce275f8c)

can you please help on this as soon as possible?

My ts version is 4.5,but still get colplied error.
/node_modules/_@floating-ui_core@1.5.1@@floating-ui/core/src/middleware/hide.d.ts(1,15): error TS1005: ',' expected.

--- a/.changeset/fuzzy-swans-doubt.md
+++ b/.changeset/fuzzy-swans-doubt.md
@@ -1,0 +1,5 @@
+---
+'@floating-ui/core': patch
+---
+
+fix(types): avoid ts 4.2+ syntax

--- a/packages/core/src/detectOverflow.ts
+++ b/packages/core/src/detectOverflow.ts
@@ -10,7 +10,7 @@ import type {
   SideObject,
 } from './types';
 
-export type Options = Partial<{
+export type DetectOverflowOptions = Partial<{
   /**
    * The clipping element(s) or area in which overflow will be checked.
    * @default 'clippingAncestors'
@@ -53,7 +53,7 @@ export type Options = Partial<{
  */
 export async function detectOverflow(
   state: MiddlewareState,
-  options: Options | Derivable<Options> = {},
+  options: DetectOverflowOptions | Derivable<DetectOverflowOptions> = {},
 ): Promise<SideObject> {
   const {x, y, platform, rects, elements, strategy} = state;
 

--- a/packages/core/src/middleware/autoPlacement.ts
+++ b/packages/core/src/middleware/autoPlacement.ts
@@ -7,10 +7,8 @@ import {
   placements as ALL_PLACEMENTS,
 } from '@floating-ui/utils';
 
-import {
-  type Options as DetectOverflowOptions,
-  detectOverflow,
-} from '../detectOverflow';
+import type {Options as DetectOverflowOptions} from '../detectOverflow';
+import {detectOverflow} from '../detectOverflow';
 import type {Alignment, Derivable, Middleware, Placement} from '../types';
 
 export function getPlacementList(

--- a/packages/core/src/middleware/autoPlacement.ts
+++ b/packages/core/src/middleware/autoPlacement.ts
@@ -7,7 +7,7 @@ import {
   placements as ALL_PLACEMENTS,
 } from '@floating-ui/utils';
 
-import type {Options as DetectOverflowOptions} from '../detectOverflow';
+import type {DetectOverflowOptions} from '../detectOverflow';
 import {detectOverflow} from '../detectOverflow';
 import type {Alignment, Derivable, Middleware, Placement} from '../types';
 

--- a/packages/core/src/middleware/hide.ts
+++ b/packages/core/src/middleware/hide.ts
@@ -1,6 +1,6 @@
 import {evaluate, sides} from '@floating-ui/utils';
 
-import type {Options as DetectOverflowOptions} from '../detectOverflow';
+import type {DetectOverflowOptions} from '../detectOverflow';
 import {detectOverflow} from '../detectOverflow';
 import type {Derivable, Middleware, Rect, SideObject} from '../types';
 

--- a/packages/core/src/middleware/hide.ts
+++ b/packages/core/src/middleware/hide.ts
@@ -1,9 +1,7 @@
 import {evaluate, sides} from '@floating-ui/utils';
 
-import {
-  type Options as DetectOverflowOptions,
-  detectOverflow,
-} from '../detectOverflow';
+import type {Options as DetectOverflowOptions} from '../detectOverflow';
+import {detectOverflow} from '../detectOverflow';
 import type {Derivable, Middleware, Rect, SideObject} from '../types';
 
 function getSideOffsets(overflow: SideObject, rect: Rect) {

--- a/packages/core/src/middleware/shift.ts
+++ b/packages/core/src/middleware/shift.ts
@@ -6,10 +6,8 @@ import {
   getSideAxis,
 } from '@floating-ui/utils';
 
-import {
-  type Options as DetectOverflowOptions,
-  detectOverflow,
-} from '../detectOverflow';
+import type {Options as DetectOverflowOptions} from '../detectOverflow';
+import {detectOverflow} from '../detectOverflow';
 import type {Coords, Derivable, Middleware, MiddlewareState} from '../types';
 
 export type ShiftOptions = Partial<

--- a/packages/core/src/middleware/shift.ts
+++ b/packages/core/src/middleware/shift.ts
@@ -6,7 +6,7 @@ import {
   getSideAxis,
 } from '@floating-ui/utils';
 
-import type {Options as DetectOverflowOptions} from '../detectOverflow';
+import type {DetectOverflowOptions} from '../detectOverflow';
 import {detectOverflow} from '../detectOverflow';
 import type {Coords, Derivable, Middleware, MiddlewareState} from '../types';
 

--- a/packages/core/src/middleware/size.ts
+++ b/packages/core/src/middleware/size.ts
@@ -7,10 +7,8 @@ import {
   min,
 } from '@floating-ui/utils';
 
-import {
-  type Options as DetectOverflowOptions,
-  detectOverflow,
-} from '../detectOverflow';
+import type {Options as DetectOverflowOptions} from '../detectOverflow';
+import {detectOverflow} from '../detectOverflow';
 import type {Derivable, Middleware, MiddlewareState} from '../types';
 
 export type SizeOptions = Partial<

--- a/packages/core/src/middleware/size.ts
+++ b/packages/core/src/middleware/size.ts
@@ -7,7 +7,7 @@ import {
   min,
 } from '@floating-ui/utils';
 
-import type {Options as DetectOverflowOptions} from '../detectOverflow';
+import type {DetectOverflowOptions} from '../detectOverflow';
 import {detectOverflow} from '../detectOverflow';
 import type {Derivable, Middleware, MiddlewareState} from '../types';
 

--- a/packages/core/src/types.ts
+++ b/packages/core/src/types.ts
@@ -184,10 +184,7 @@ export type RootBoundary = 'viewport' | 'document' | Rect;
 export type ElementContext = 'reference' | 'floating';
 
 export {computePosition} from './computePosition';
-export {
-  detectOverflow,
-  Options as DetectOverflowOptions,
-} from './detectOverflow';
+export {detectOverflow, DetectOverflowOptions} from './detectOverflow';
 export {arrow, ArrowOptions} from './middleware/arrow';
 export {autoPlacement, AutoPlacementOptions} from './middleware/autoPlacement';
 export {flip, FlipOptions} from './middleware/flip';

